src/main/java/com/pingcap/tikv/statistics/Histogram.java

```diff
@@ -0,0 +1,209 @@
+ * limitations under the License.
+ */
+package com.pingcap.tikv.statistics;
+
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.ByteString;
+import com.pingcap.tikv.Snapshot;
+import com.pingcap.tikv.TiCluster;
+import com.pingcap.tikv.TiConfiguration;
+import com.pingcap.tikv.catalog.Catalog;
+import com.pingcap.tikv.expression.TiColumnRef;
+import com.pingcap.tikv.expression.TiConstant;
+import com.pingcap.tikv.expression.TiExpr;
+import com.pingcap.tikv.expression.scalar.Equal;
+import com.pingcap.tikv.meta.TiDBInfo;
+import com.pingcap.tikv.meta.TiIndexInfo;
+import com.pingcap.tikv.meta.TiSelectRequest;
+import com.pingcap.tikv.meta.TiTableInfo;
+import com.pingcap.tikv.operation.SchemaInfer;
+import com.pingcap.tikv.predicates.PredicateUtils;
+import com.pingcap.tikv.predicates.ScanBuilder;
+import com.pingcap.tikv.row.Row;
+import com.pingcap.tikv.types.DataTypeFactory;
+import com.pingcap.tikv.types.Types;
+import com.pingcap.tikv.util.Bucket;
+import com.pingcap.tikv.util.Comparables;
+import com.pingcap.tikv.util.RangeSplitter;
+
+import java.lang.reflect.Type;
+import java.util.Iterator;
+import java.util.List;
+
+
+public class Histogram {
+
+    private final long distinctValueOfNumber = 0;
+    public Bucket[] buckets;
+
+    private static TiConfiguration conf = TiConfiguration.createDefault(ImmutableList.of("127.0.0.1:" + 2379));
+    private static TiCluster cluster = TiCluster.getCluster(conf);
+    private static Snapshot snapshot = cluster.createSnapshot();
+
+    // histogramFromStorage from the storage to histogram.
+    public void histogramFromStorage(long tableID,long isIndex,long colID){
+        Catalog cat = cluster.getCatalog();
+        TiDBInfo db = cat.getDatabase("mysql");
+        TiTableInfo table = cat.getTable(db, "stats_buckets");
+        TiIndexInfo index = TiIndexInfo.generateFakePrimaryKeyIndex(table);
+
+
+        //"select bucket_id, count, repeats, lower_bound, upper_bound from mysql.x
+    // where table_id = %d and is_index = %d and hist_id = %d", tableID, isIndex, colID
+        List<TiExpr> firstAnd = ImmutableList.of(
+                new Equal(TiColumnRef.create("table_id", table), TiConstant.create(tableID)),
+                new Equal(TiColumnRef.create("is_index", table), TiConstant.create(isIndex)),
+                new Equal(TiColumnRef.create("hist_id", table), TiConstant.create(colID))
+        );
+
+        ScanBuilder scanBuilder = new ScanBuilder();
+        ScanBuilder.ScanPlan scanPlan = scanBuilder.buildScan(firstAnd, index, table);
+
+        TiSelectRequest selReq = new TiSelectRequest();
+        selReq.addRanges(scanPlan.getKeyRanges())
+                .setTableInfo(table)
+                .setIndexInfo(index)
+                .addField(TiColumnRef.create("table_id", table))
+                .addField(TiColumnRef.create("is_index", table))
+                .addField(TiColumnRef.create("hist_id", table))
```

Comment:
ok  when double read PR merged ,i will uodate ``` addField```  to ```addRequiredColumn```.I will pay close attention to the PR at any time